rest/handler/tracinghandler.go

```diff
@@ -48,7 +72,21 @@ func TracingHandler(serviceName, path string) func(http.Handler) http.Handler {
 
 			// convenient for tracking error messages
 			propagator.Inject(spanCtx, propagation.HeaderCarrier(w.Header()))
-			next.ServeHTTP(w, r.WithContext(spanCtx))
+			trw := &traceResponseWriter{
+				w:    w,
+				code: http.StatusOK,
+			}
+			next.ServeHTTP(trw, r.WithContext(spanCtx))
+
+			span.SetAttributes(attribute.KeyValue{
+				Key:   traceKeyStatusCode,
+				Value: attribute.IntValue(trw.code),
+			})
+			if trw.code >= http.StatusInternalServerError {
+				span.SetStatus(codes.Error, http.StatusText(trw.code))
```

Comment:
Why only keep codes >= 500?

Reply:
Update to >= 400, 4xx is also an abnormal condition and requires a warning.